ding/framework/parallel.py

```diff
@@ -56,6 +56,10 @@ def _run(
         self._listener = Thread(target=self.listen, name="mq_listener", daemon=True)
         self._listener.start()
 
+        self.mq_type = mq_type
+        from ding.framework.middleware.barrier import BarrierRuntime
```

Comment:
move import location

Reply:
Something tricky here, we have to get the BarrierRuntime object in the closure to avoid circular import.

Reply:
ok, add a line comment about this